### DISCUSSION TITLE
How use prior targets as input values for tar_map?

### DISCUSSION THREAD

**Discussion by wmoldham** [ropensci/tarchetypes#91] [Help] [answered]
I am trying to chain together two `tar_map` functions and was wondering if it was possible to use a subset of the targets generated by the first `tar_map` as the `values` for the second? Here is a sketch of what I'm trying to accomplish:

```r
    list(
      tar_map(
        values = list(x = 1:8), 
        names = x, 
        tar_target(number, x)
      ), 
      tar_map(
        values = list(
          first = c(number_1, number_3), 
          second = c(number_2, number_4)
        ), 
        tar_target(
          added, 
          sum(first, second)
        )
      )
    )
```

Note that I only need to analyze a subset of the results from the first `tar_map` function. My expected output would be two new targets (added) that are the sum of 1+2 and 3+4. Unfortunately, I can't seem to figure out how to pass `number_1`, etc. into the second `tar_map` function of if this is even possible. Currently, I am writing out separate targets for each of the combinations, and so it would be handy to have a programmatic way to use `tar_map` here. Thanks for any assistance!

**Comment by wlandau** [comment, marked as answer]
`tar_map()` works on [language objects](https://adv-r.hadley.nz/expressions.html), so it should work if you write `values = list(first = rlang::syms(c("number_1", "number_2")), second = rlang::syms(c("number_2", "number_4")))`.

**Reply by wmoldham** [reply]
Many thanks! While I had tried that, I apparently forgot that the `names` argument for `tar_map` refers to the name of one of the `values` and not a literal string of names! Fixing both solved my issue.
